airflow/api/common/experimental/remove_dag.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from airflow import models, settings
+from airflow.exceptions import AirflowException
+from sqlalchemy import or_
+
+
+def remove_dag(dag_id):
+    count = 0
+
+    session = settings.Session()
+    DM = models.DagModel
+    dag = session.query(DM).filter(DM.dag_id==dag_id).first()
```

Comment:
You probably want to check if the dag exists in the DagBag as well. If it does, it will get re-added which could lead to some race conditions.